### PR TITLE
travis-ci: add build for Ubuntu 18.04 and drop Ubuntu 12.04

### PR DESCRIPTION
**Pull request by AlexandraTrifan**
- Add a docker build for Ubuntu 18.04.1 LTS (Bionic Beaver) : EOL April 2023.
- Ubuntu 12.04.5 LTS reached its regular End of Life on April 28, 2017 so we should remove the corresponding Travis CI build. 
- Builds with Xcode 6.4 in Travis CI are deprecated and will be removed in January 2019, so we can remove the Xcode 6.4 builds from Travis CI.

Signed-off-by: Alexandra.Trifan <Alexandra.Trifan@analog.com>

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -36,9 +36,6 @@ matrix:
       os: linux
       dist: trusty
       env: LDIST=DO_NOT_DEPLOY
-    - compiler: "gcc"
-      os: linux
-      dist: precise
```
Comment (commodo) [inline .travis.yml]: I think you could also remove `osx_image: xcode6.4`
you could do it in this commit, or make a new one ; doesn't matter much

File: CI/travis/before_install_linux [hunk 2]
```diff
@@ -11,6 +11,13 @@ handle_centos() {
 	sudo docker pull centos:centos${OS_VERSION}
 }
 
+handle_bionic() {
```
Comment (commodo) [inline CI/travis/before_install_linux]: you could call this `handle_ubuntu_docker` and it should only run for builds that define `OS_TYPE=ubuntu_docker`
and then you can do `sudo docker pull ubuntu:$OS_VERSION` and you wouldn't need the changes in the `handle_${OS_TYPE}` logic

File: cmake/LinuxPackaging.cmake [hunk 3]
```diff
@@ -48,6 +48,9 @@ if(DEB_DETECT_DEPENDENCIES AND DPKG_CMD AND DPKGQ_CMD)
 	if(WITH_XML_BACKEND)
 		set(PACKAGES "${PACKAGES} libxml2")
 	endif()
+	if(WITH_SERIAL_BACKEND)
```
Comment (commodo) [inline cmake/LinuxPackaging.cmake]: these packaging changes look like they could go into a separate commit that handles only Linux packaging

File: CI/travis/inside_bionic_docker.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
+#!/bin/sh -xe
+
+TRAVIS_CI="$1"
+
+apt-get -qq update
+apt-get install -y cmake doxygen libaio-dev libavahi-client-dev libavahi-common-dev libusb-1.0-0-dev libxml2-dev git
+apt-get install -y tar bzip2 gzip flex bison
```
Comment (commodo) [inline CI/travis/inside_bionic_docker.sh]: you could include the `before_install_linux` script here ;
for Ubuntu this is do-able, since it's common code with the main code that runs in Travis-CI

File: CI/travis/make_linux [hunk 5]
```diff
@@ -16,8 +16,19 @@ handle_centos() {
 		/bin/bash -xe /libiio/CI/travis/inside_centos_docker.sh ${OS_VERSION} travis-ci
 }
 
+handle_bionic() {
```
Comment (commodo) [inline CI/travis/make_linux]: same here with `handle_ubuntu_docker` and using `$OS_VERSION` instead of `bionic`

this helps us extend for future Ubuntu versions (if we want to)

**Comment by commodo** [comment]
Codacy has one small complaint.
I admit that this code was adapted from an existing one; we could fix the old one as well (not necessarily in this PR).

File: .travis.yml (hunk 1 above)
Comment (AlexandraTrifan) [reply .travis.yml]: Added a new commit to remove the xcode 6.4 image

**Comment by commodo** [comment]
The Codacy complaint went away after `before_install_linux` was included;

File: .travis.yml [hunk 6]
```diff
@@ -36,9 +36,6 @@ matrix:
       os: linux
       dist: trusty
       env: LDIST=DO_NOT_DEPLOY
-    - compiler: "gcc"
```
Comment (commodo) [inline .travis.yml]: for this change it would be a good idea to add a small comment about why it's being removed (i.e. 12.04 has been deprecated, and will be dropped by Travis-CI in the future, etc, etc, etc)

File: .travis.yml [hunk 7]
```diff
@@ -56,9 +56,6 @@ matrix:
       env:
         - OS_TYPE=ubuntu_docker
         - OS_VERSION=bionic
-    - compiler: "gcc"
```
Comment (commodo) [inline .travis.yml]: similarly:
for this change it would be a good idea to add a small comment about why it's being removed (i.e. Xcode 6.4 has been deprecated, and will be dropped by Travis-CI in the future, etc, etc, etc)

**Comment by commodo** [comment]
[LGTM](http://i.imgur.com/YuL9Yns.jpg)

File: .travis.yml (hunk 6 above)
Comment (AlexandraTrifan) [reply .travis.yml]: Added some details in the commit message and the PR description.

File: .travis.yml (hunk 7 above)
Comment (AlexandraTrifan) [reply .travis.yml]: the same: added some details in the commit message and the PR description.

**Comment by tfcollins** [comment]
Those three look fine to me.  Ideally, we would have 8.2 but that isn't available. Drop the oldest if we want to reduce the number of builds.

**Comment by AlexandraTrifan** [comment]
The Xcode 8.1 build is automatically routed to the Xcode 8.3 image.
I will make the required changes to use those three versions and I will create a new pull request for that, if that's ok.

**Comment by commodo** [comment]
Let's do in a subsequent PR.
It allows us to move forward and iterate over this and other takes.
